src/components/Modal/index.tsx

```diff
@@ -1,69 +1,255 @@
-  }['bivarianceHack']
-  modalClassName?: string
+  handleClose?: (event: Record<string, unknown>, reason: 'backdropClick' | 'escapeKeyDown') => void
   open: boolean
   paperClassName?: string
   title: string
 }
 
-const GnoModal = ({
-  children,
-  description,
-  handleClose,
-  modalClassName,
-  open,
-  paperClassName,
-  title,
-}: GnoModalProps): ReactElement => {
-  const classes = useStyles()
-
+const GnoModal = ({ children, description, handleClose, open, paperClassName, title }: GnoModalProps): ReactElement => {
   return (
-    <Modal
+    <ModalStyled
+      BackdropProps={{ className: 'overlay' }}
       aria-describedby={description}
       aria-labelledby={title}
-      className={cn(classes.root, modalClassName)}
       onClose={handleClose}
       open={open}
     >
-      <div className={cn(classes.paper, paperClassName, 'classpep')}>{children}</div>
-    </Modal>
+      <div className={cn('paper', paperClassName)}>{children}</div>
+    </ModalStyled>
   )
 }
 
 export default GnoModal
+
+/*****************/
+/* Generic Modal */
+/*****************/
+
+/*** Header ***/
+const HeaderSection = styled.div`
+  display: flex;
+  padding: 24px 18px 24px 24px;
+  border-bottom: 2px solid ${({ theme }) => theme.colors.separator};
+
+  h5 {
+    color: ${({ theme }) => theme.colors.text};
+  }
+
+  .close-button {
+    align-self: flex-end;
+    background: none;
+    border: none;
+    padding: 5px;
+    width: 26px;
+    height: 26px;
+
+    span {
+      margin-right: 0;
+    }
+
+    :hover {
+      background: ${({ theme }) => theme.colors.separator};
+      border-radius: 16px;
+      cursor: pointer;
+    }
+  }
+`
+
+const TitleStyled = styled(TitleSRC)`
+  display: flex;
+  align-items: center;
+  flex-basis: 100%;
+
+  .image,
+  img {
+    width: 20px;
+    margin-right: 10px;
+  }
+
+  .note,
+  span {
+    margin-left: 12px;
+  }
+`
+
+interface TitleProps {
+  children: string | ReactNode
+  size?: keyof Theme['title']['size']
+  withoutMargin?: boolean
+  strong?: boolean
+}
+
+const Title = ({ children, ...props }: TitleProps): ReactElement => (
+  <TitleStyled size="xs" withoutMargin {...props}>
+    {children}
+  </TitleStyled>
+)
+
+interface HeaderProps {
+  children?: ReactNode
+  onClose?: (event: any) => void
+}
+
+const Header = ({ children, onClose }: HeaderProps): ReactElement => {
+  return (
+    <HeaderSection className="modal-header">
+      {children}
+
+      {onClose && (
+        <button className="close-button" onClick={onClose}>
+          <Icon size="sm" type="cross" />
+        </button>
+      )}
+    </HeaderSection>
+  )
+}
+
+Header.Title = Title
+
+/*** Body ***/
+const BodySection = styled.div<{ withoutPadding: BodyProps['withoutPadding'] }>`
+  padding: ${({ withoutPadding }) => (withoutPadding ? 0 : '24px')};
+`
+
+interface BodyProps {
+  children: ReactNode | ReactNodeArray
+  withoutPadding?: boolean
```

Comment:
perhaps this is against designs, but wdyt about changing it to `padding?: number` if it's undefined, we set it to 24. 
It could give more flexibility.

Reply:
mmm nevermind, it's not a good idea.